src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -472,8 +446,8 @@ async ValueTask<ulong> Complete(bool async, CancellationToken cancellationToken
             }
             catch
             {
-                // An exception here will have already broken the connection etc.
-                Cleanup();
+                if (_state != ImporterState.Disposed)
+                    Cleanup();
```

Comment:
Maybe just add a check inside Cleanup which exits immediately if disposed?

Reply:
We can, but the idea here was for the `Cleanup` to be the same across every COPY operation.

Reply:
Well, they're different methods in different classes (with slight differences, because things work different across import/export), so I'm not sure consistency is actually important here...

But up to you.